test/functional/test_framework/v2_p2p.py

```diff
@@ -4,10 +4,92 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 """Class for v2 P2P protocol (see BIP 324)"""
 
-from .crypto.ellswift import ellswift_ecdh_xonly
+import logging
+import os
+import random
+
+from .crypto.bip324_cipher import FSChaCha20Poly1305
+from .crypto.chacha20 import FSChaCha20
+from .crypto.ellswift import ellswift_create, ellswift_ecdh_xonly
+from .crypto.hkdf import hkdf_sha256
 from .key import TaggedHash
 
+logger = logging.getLogger("TestFramework.v2_p2p")
+
+MAGIC_BYTES = {
+    "regtest": b"\xfa\xbf\xb5\xda"   # regtest
+}
```

Comment:
I agree that would be nice, in particular so that we can enable `p2p_dos_header_tree.py` (which uses testnet instead of regtest) in combination with v2 transport. One possibility would be to move the `MAGIC_BYTES` constants from `p2p.py` into another file (e.g. `test_framework/messages.py`), change all includes, pass `net` to `EncryptedP2PState`, and change `V1_PREFIX` into a local variable using it)

Reply:
nice! included `MAGIC_BYTES` in messages.py

Reply:
Thanks! 
I think you should actually use `net` by making all callers from `p2p.py`  pass it to `EncryptedP2PState`, the default value of 'regtest' should be dropped in my opinion. You can verify that it works when `p2p_dos_header_tree.py --transport` doesn't fail anymore.

Could also change all existing imports for MAGIC_BYTES (there are a few in other tests) to avoid indirect imports.

Reply:
sounds much better! added a [new commit](https://github.com/bitcoin/bitcoin/pull/24748/commits/7f0e797dc9524c03f650f60a028cb46852877284).